python/contrib/dag/__init__.py

```diff
@@ -79,18 +99,28 @@ def add_edge(self, ind_node, dep_node, graph=None):
             self.delete_edge(ind_node, dep_node)
             raise DAGValidationError()
 
+    # Modified by the MOOSE group
     def delete_edge(self, ind_node, dep_node, graph=None):
         """ Delete an edge from the graph. """
         if not graph:
             graph = self.graph
+
+        # save original graph
+        self.__createOriginalGraph()
```

Comment:
`cacheGraph` is a good name - especially since it has nothing to do with "original"

Reply:
cool, will do